server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java

```diff
@@ -182,15 +189,24 @@ Map<Collector, Map<String, Long>> buildSliceLevelBreakdown(long createWeightStar
                     );
                 }
                 // compute sliceMaxEndTime as max of sliceEndTime across all timing types
-                sliceMaxEndTime = Math.max(sliceMaxEndTime, currentSliceBreakdown.get(timingTypeSliceEndTimeKey));
+                sliceMaxEndTime = Math.max(sliceMaxEndTime, currentSliceBreakdown.getOrDefault(timingTypeSliceEndTimeKey, Long.MIN_VALUE));
+                sliceMinStartTime = Math.min(
+                    sliceMinStartTime,
+                    currentSliceBreakdown.getOrDefault(timingTypeSliceStartTimeKey, Long.MAX_VALUE)
+                );
                 // compute total time for each timing type at slice level using sliceEndTime and sliceStartTime
                 currentSliceBreakdown.put(
                     timingType.toString(),
-                    currentSliceBreakdown.get(timingTypeSliceEndTimeKey) - currentSliceBreakdown.get(timingTypeSliceStartTimeKey)
+                    currentSliceBreakdown.getOrDefault(timingTypeSliceEndTimeKey, 0L) - currentSliceBreakdown.getOrDefault(
+                        timingTypeSliceStartTimeKey,
+                        0L
+                    )
                 );
             }
-            // currentSliceNodeTime includes the create weight time as well which will be same for all the slices
-            long currentSliceNodeTime = sliceMaxEndTime - createWeightStartTime;
+            // currentSliceNodeTime does not include the create weight time, as that is computed in non-concurrent part
+            long currentSliceNodeTime = (sliceMaxEndTime == Long.MIN_VALUE && sliceMinStartTime == Long.MAX_VALUE)
```

Comment:
We probably should use || here to make sure both values are properly calculated:

```suggestion
            long currentSliceNodeTime = (sliceMaxEndTime == Long.MIN_VALUE || sliceMinStartTime == Long.MAX_VALUE)
```

Reply:
@reta May be I can add an assert before this that we expect both the variables to be set with valid values or both to  have default values.

Reply:
Sure, please be aware the assertions could be turned off, so you still have to properly check the conditions